File: examples/cli-batch-python/python/src/cli_batch_demo/batch.py
"""Line-oriented batch transforms."""

from __future__ import annotations

from collections.abc import Callable, Iterable


def map_lines(lines: Iterable[str], fn: Callable[[str], str]) -> list[str]:
    """Apply ``fn`` to each non-empty stripped line."""
    out: list[str] = []
    for raw in lines:
        line = raw.rstrip("\n")
        if not line.strip():
            continue
        out.append(fn(line))
    return out

File: examples/cli-batch-python/kotlin/src/main/kotlin/dev/pykotmig/clibatch/Batch.kt
package dev.pykotmig.clibatch

fun mapLines(
    lines: Iterable<String>,
    transform: (String) -> String,
): List<String> {
    val out = ArrayList<String>()
    for (raw in lines) {
        val line = raw.trimEnd('\n')
        if (line.isBlank()) {
            continue
        }
        out.add(transform(line))
    }
    return out
}
